cli/tools/upgrade.rs

```diff
@@ -121,58 +82,43 @@ pub async fn upgrade_command(
     }
   }
 
-  println!("Upgrade done successfully");
+  println!("Upgraded successfully");
 
   Ok(())
 }
 
-fn download_package(
-  url: &Url,
-  client: Client,
-  version: &Version,
-) -> Pin<Box<dyn Future<Output = Result<Vec<u8>, AnyError>>>> {
-  println!("downloading {}", url);
-  let url = url.clone();
-  let version = version.clone();
-  let fut = async move {
-    match fetch_once(client.clone(), &url, None).await {
-      Ok(result) => {
-        println!(
-          "Version has been found\nDeno is upgrading to version {}",
-          &version
-        );
-        match result {
-          FetchOnceResult::Code(source, _) => Ok(source),
-          FetchOnceResult::NotModified => unreachable!(),
-          FetchOnceResult::Redirect(_url, _) => {
-            download_package(&_url, client, &version).await
-          }
-        }
-      }
-      Err(_) => {
-        println!("Version has not been found, aborting");
-        std::process::exit(1)
-      }
-    }
-  };
-  fut.boxed_local()
+async fn get_latest_version(client: &Client) -> Result<String, AnyError> {
+  println!("Looking up latest version");
+
+  let res = client
+    .get(&format!("{}/latest", RELEASE_URL))
+    .send()
+    .await?;
+  let version = res.url().path_segments().unwrap().last().unwrap();
+
+  Ok(version.replace("v", ""))
 }
 
-fn compose_url_to_exec(version: &Version) -> Result<Url, AnyError> {
-  let s = format!(
-    "https://github.com/denoland/deno/releases/download/v{}/{}",
-    version, *ARCHIVE_NAME
+async fn download_package(
+  client: Client,
+  install_version: &str,
+) -> Result<Vec<u8>, AnyError> {
+  let download_url = format!(
+    "{}/download/v{}/{}",
+    RELEASE_URL, install_version, *ARCHIVE_NAME
   );
-  Url::parse(&s).map_err(AnyError::from)
-}
 
-fn find_version(text: &str) -> Result<String, AnyError> {
-  let re = Regex::new(r#"v([^\?]+)?""#)?;
-  if let Some(_mat) = re.find(text) {
-    let mat = _mat.as_str();
-    return Ok(mat[1..mat.len() - 1].to_string());
+  println!("Checking {}", &download_url);
+
+  let res = client.get(&download_url).send().await?;
+
+  if res.status().is_success() {
+    println!("Download has been found");
```

Comment:
`Version has been found`

Reply:
This will seem very write in the complete execution of the command though:
```
Looking up latest version
Found latest version 1.5.3
Checking https://github.com/denoland/deno/releases/download/v1.5.3/deno-x86_64-apple-darwin.zip
Download has been found
Deno is upgrading to version 1.5.3
Archive:  /var/folders/mb/xj2kfz795sj87yhhzd8p0f0c0000gn/T/.tmpNSqBtr/deno.zip
  inflating: deno
Upgraded successfully
```
would turn into
```
Looking up latest version
Found latest version 1.5.3
Checking https://github.com/denoland/deno/releases/download/v1.5.3/deno-x86_64-apple-darwin.zip
Version has been found
Deno is upgrading to version 1.5.3
Archive:  /var/folders/mb/xj2kfz795sj87yhhzd8p0f0c0000gn/T/.tmpNSqBtr/deno.zip
  inflating: deno
Upgraded successfully
```

`Found latest version` and then `Version has been found`

Reply:
Okay, makes sense then